docs/migratation-guide-v1.0.rst

```diff
@@ -0,0 +1,311 @@
+good test coverage before upgrading.
+
+Most applications applications using a version < 1.0.0 should continue
+to work but it’s still highly recommended to test your application
+first.
+
+.. WARNING::
+   **BREAKING CHANGES**
+
+BREAKING CHANGES
+================
+
+``iConn.run()`` no longer supports sync mode
+--------------------------------------------
+
+Sync mode did not work properly to begin with see
+(`#32 <https://github.com/IBM/nodejs-itoolkit/issues/32>`__).
+
+``iConn.setTimeout()`` is removed
+----------------------------------
+
+This function was used to set timeout for ``iConn.run`` sync mode.
+
+``iSql.connect()`` and ``iSql.setOptions()`` are removed
+--------------------------------------------------------
+
+These functions were used in conjunction for XMLSERVICE user
+authentication. The transports already handle user authentication.
+
+New Features
+============
+
+SSH and ODBC transports
+-----------------------
+
+Users can now use ``ssh`` and ``odbc`` transports. This will allow users
+more ways to use itoolkit on their local machine.
+
+Support error first callbacks
+-----------------------------
+
+-  ``iConn.run()`` did not return errors to the run callback.
+   ``Connection.run()`` follows Node.js convention of `error first
+   callbacks <https://nodejs.org/api/errors.html#errors_error_first_callbacks>`__.
+   ``Connection`` still has a compatability option ``returnError`` to
+   behave like ``iConn.run()`` and return the xml output as the first
+   parameter of the run callback.
+
+
+Support DS types within addReturn
+---------------------------------
+
+-  ``iPgm.addReturn`` did not support the DS data type,
+   ``ProgramCall.addReturn`` added support.
+
+Deprecated Classes and Functions
+================================
+
+iConn
+-----
+
+The ``iConn`` class was replaced with the ``Connection`` class and it
+will be removed in ``v2.x``.
+
+Differences
+^^^^^^^^^^^
+
+-  ``Connection`` constructor accepts single object parameter.
+-  ``Connection.run()`` follows Node.js convention of error first `error
+   first
+   callbacks <https://nodejs.org/api/errors.html#errors_error_first_callbacks>`__.
+
+Migrating from ``iConn`` to ``Connection`` Constructor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: js
+
+   // using idb transport
+
+   // const conn = new iConn("*LOCAL", "myuser", "mypassword");
+
+   const conn = new Connection({
+     transport: 'idb',
+     transportOptions: { database: '*LOCAL', username: 'myuser', password: 'mypassword' }
+   });
+
+   // using rest transport
+
+   // const restConfig = { host: 'myhost', port: 80, path: '/' }
+   // const conn = new iConn('*LOCAL', 'myuser', 'mypassword', restConfig);
+
+   const conn = new Connection({
+     transport: 'rest',
+     transportOptions: {
+       database: '*LOCAL',
+       username: 'myuser',
+       password: 'mypassword',
+       url: 'http://myhost:80/',
```

Comment:
```suggestion
       url: 'http://myhost.example.com/cgi-bin/xmlcgi.pgm',
```

Not sure how we got to documenting / as the default endpoint when all the Apache docs we have set it up as /cgi-bin/xmlcgi.pgm.